### PR TITLE
test: Update combine tests to support gen2 style relationships

### PR DESCRIPTION
**Pull request by palpatim**
#### Description of changes

Updates combine unit & synthesis tests to validate gen2-style relationships (using `references`)

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by atierian — APPROVED** [review]
One thought:
By removing `fails to combine heterogeneous related definitions for multiple supported db types`, we no have a negative test for that case. Should we?

Otherwise LGTM.

File: packages/amplify-graphql-api-construct/src/__tests__/__functional__/auth-modes.test.ts [hunk 1]
```diff
@@ -121,6 +122,100 @@ describe('auth modes', () => {
...
+        authorizationModes: {
+          defaultAuthorizationMode: 'AMAZON_COGNITO_USER_POOLS',
+          apiKeyConfig: { expires: cdk.Duration.days(7) },
+          userPoolConfig: { userPool },
+        },
+      });
+    });
+  });
+
+  it('renders with user pool auth across data sources: ddb primary, sql related', () => {
```
Comment (palpatim) [inline packages/amplify-graphql-api-construct/src/__tests__/__functional__/auth-modes.test.ts:172]: It probably shouldn't matter which side of the relationship has which data source, but better safe than sorry

File: packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts [hunk 2]
```diff
@@ -732,7 +733,7 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
           ...acm.getRolesPerOperation('listen'),
         ]),
       ].map((r) => this.roleMap.get(r)!);
-      relatedAuthExpression = this.getVtlGenerator(ctx, def.name.value).generateAuthExpressionForRelationQuery(
+      relatedAuthExpression = this.getVtlGenerator(ctx, relatedModelName).generateAuthExpressionForRelationQuery(
```
Comment (palpatim) [inline packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts:736]: This is the change that supports auth across relationships amongst heterogeneous data sources

**Comment by palpatim** [comment]
> One thought: By removing `fails to combine heterogeneous related definitions for multiple supported db types`, we no have a negative test for that case. Should we?
> 
> Otherwise LGTM.

Will add in an upcoming PR

**Review by phani-srikar — APPROVED** [review]

**Review by atierian — APPROVED** [review]

**Review by iartemiev — APPROVED** [review]

**Pull request merged**